lib/matplotlib/backend_bases.py

```diff
@@ -1490,10 +1533,18 @@ def on_key(event):
 
         cid = fig.canvas.mpl_connect('key_press_event', on_key)
     """
+
     def __init__(self, name, canvas, key, x=0, y=0, guiEvent=None):
         LocationEvent.__init__(self, name, canvas, x, y, guiEvent=guiEvent)
         self.key = key
 
+    def process(self):
```

Comment:
This (or the init) should update the location based on the last-known mouse location if the user does not supply one.

Reply:
Did that occur before?  One of the points of this PR is also to stop trying to fill in this "bad" metadata, which can easily be wrong.  (If it did occur, sure, I can put that back for now, but I don't think it did?)

Reply:
That is my understanding of https://github.com/matplotlib/matplotlib/blob/e9e1c876a5ee9f327c2cece2808698d11801fa13/lib/matplotlib/backend_bases.py#L1691-L1692 and https://github.com/matplotlib/matplotlib/blob/e9e1c876a5ee9f327c2cece2808698d11801fa13/lib/matplotlib/backend_bases.py#L1700-L1701